Creates EquippableItem

kamronbatman opened the pull request:
Creates EquippableItem and adds Containers, Light sources, and Equippables to it.
Removes Customs Framework.

jaedan commented:
If you're going to do a code style change, do it as a separate pull request just for that purpose.

jaedan commented:
Consider using something like this to enforce style too https://blogs.msdn.microsoft.com/dotnet/2016/12/15/code-style-configuration-in-the-vs2017-rc-update/

kamronbatman commented:
Unfortunately this PR is bleeding.  I removed the CustomFrameworks because there is a List<Module> on every item/Mobile.  And the Server World/Serialization is intertwined with CustomSerial/SaveData, which not everyone needs.  It should not have been implemented and since it was preventing a clean PR, I just removed it.

As for the style changes, it is automatic in Visual Studio 2017.  Preventing it is more work. :(

kamronbatman commented:
@jaedan - I think I am going to break this PR up before merging.  Because I need to get roslyn working.  Then I want to remove the customs framework.  Then I want to force styling so I don't have these formatting issues.  Then I'll do the Layer -> EquippableItem.

Thoughts?
